File: String/Minimum indexed character/minIndexed.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  int minIndexChar(string str, string patt) {
    int n = patt.length();
    int res = INT_MAX;
    int count[n] = {0};

    for (int i = 0; i < n; i++) {
      count[i] = str.find(patt[i]);
      if (count[i] >= 0) res = min(res, count[i]);
    }

    for (int i = 0; i < n; i++) {
      if (count[i] == -1) {
        if (res == INT_MAX)
          res = min(res, count[i]);
        else
          res = max(res, count[i]);
      }
    }
    return res;
  }
};

int main() {
  int t;
  cin >> t;
  while (t--) {
    string str;
    string patt;
    cin >> str;
    cin >> patt;
    Solution obj;
    cout << obj.minIndexChar(str, patt) << endl;
  }
  return 0;
}

File: String/Minimum indexed character/MinIndexed.java
import java.util.ArrayList;
import java.util.Scanner;

class Solution {
  public static int minIndexChar(String str, String patt) {
    int l1 = str.length(), l2 = patt.length();
    ArrayList<Character> s = new ArrayList<Character>();
    for (int x = 0; x < l2; x++) {
      if (!s.contains(patt.charAt(x))) s.add(patt.charAt(x));
    }
    for (int x = 0; x < l1; x++) {
      if (s.contains(str.charAt(x))) return x;
    }
    return -1;
  }
}

class MinIndexed {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    while (t > 0) {
      t--;
      String s1 = sc.next();
      String s2 = sc.next();
      int res = minIndexChar(s1, s2);
      System.out.println(res);
    }
  }
}
